pkg/services/app/commit_test.go

```diff
@@ -0,0 +1,98 @@
+package app
+
+import (
+	"time"
+
+	"github.com/fluxcd/go-git-providers/gitprovider"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	wego "github.com/weaveworks/weave-gitops/api/v1alpha1"
+	"github.com/weaveworks/weave-gitops/pkg/gitproviders"
+)
+
+var _ = Describe("Get Commits", func() {
+	It("gets commits for a user app", func() {
+		commitParams := CommitParams{
+			Name:      "test",
+			Namespace: "wego-system",
+		}
+
+		application := &wego.Application{
+			Spec: wego.ApplicationSpec{URL: "https://github.com/foo/bar"},
+		}
+
+		gitProviders.GetAccountTypeStub = func(s string) (gitproviders.ProviderAccountType, error) {
+			return gitproviders.AccountTypeUser, nil
+		}
+
+		commits := []gitprovider.Commit{&fakeCommit{}}
+		gitProviders.GetCommitsFromUserRepoStub = func(gitprovider.UserRepositoryRef, string, int, int) ([]gitprovider.Commit, error) {
+			return commits, nil
+		}
+
+		commit, err := appSrv.GetCommits(commitParams, application)
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(len(commit)).To(Equal(1))
+	})
+
+	It("gets commits for a user app", func() {
+		commitParams := CommitParams{
+			Name:      "test",
+			Namespace: "wego-system",
+		}
+
+		application := &wego.Application{
+			Spec: wego.ApplicationSpec{URL: "https://github.com/foo/bar"},
+		}
+
+		gitProviders.GetAccountTypeStub = func(s string) (gitproviders.ProviderAccountType, error) {
+			return gitproviders.AccountTypeOrg, nil
+		}
+
+		commits := []gitprovider.Commit{&fakeCommit{}}
+		gitProviders.GetCommitsFromOrgRepoStub = func(gitprovider.OrgRepositoryRef, string, int, int) ([]gitprovider.Commit, error) {
+			return commits, nil
+		}
+
+		commit, err := appSrv.GetCommits(commitParams, application)
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(len(commit)).To(Equal(1))
+	})
+
+	It("fails to get commits for helm", func() {
+		commitParams := CommitParams{
+			Name:      "test",
+			Namespace: "wego-system",
+		}
+
+		application := &wego.Application{
+			Spec: wego.ApplicationSpec{SourceType: wego.SourceTypeHelm},
+		}
+
+		_, err := appSrv.GetCommits(commitParams, application)
+		Expect(err).Should(HaveOccurred())
+		Expect(err.Error()).Should(Equal("unable to get commits for a helm chart"))
```

Comment:
Can we use an error type here instead of looking for an exact string match? If someone wraps the error or changes one character, the test fails.

Reply:
I will follow what I see in other places just checking if an err occured.